src/Feature/SignatureHelp/Feature_SignatureHelp.re

```diff
@@ -523,17 +529,19 @@ module View = {
             />
           </View>
         </UI.Components.Row>
-        {switch (parameter.documentation) {
-         | Some(docs) when Exthost.MarkdownString.toString(docs) != "" =>
+        {switch (maybeParameter) {
+         | Some({documentation: Some(docs), _})
+             when Exthost.MarkdownString.toString(docs) != "" =>
            [
              <horizontalRule theme=colorTheme />,
              <signatureHelpMarkdown markdown=docs />,
            ]
            |> React.listToElement
          | _ => React.empty
          }}
```

Comment:
You could perhaps decrease nesting by using monadic bind / flatmap, ie:
```
maybeParameter
|> OptionEx.flatMap(({documentation, _}: Parameter.t) => documentation)
|> Option.map(docs => {
    if (Exthost.MarkdownString.toString(docs)) == 0 ? React.empty : 
           [
              <horizontalRule theme=colorTheme />,
             <signatureHelpMarkdown markdown=docs />,
           ]
           |> React.listToElement;
}
|> Option.value(~default=React.empty);
```
Not a big deal here, though, the delta is pretty small.. but a good tool to keep in mind to flatten nested switches / pattern matches.